Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -197,92 +213,18 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
         )
       }
 
-    // MARK: ApplePay
-
-    /*
+    // MARK: Apple Pay
 
-     Order of operations:
-     1) Apple pay button tapped
-     2) Generate a new payment intent
-     3) Present the payment authorization form
-     4) Payment authorization form calls applePayContextDidCreatePayment with the payment method Id
-     5) Payment authorization form calls paymentAuthorizationDidFinish
-     6) Validate checkout using the checkoutId, payment source id, and payment intent
-     */
-
-    let createPaymentIntentForApplePay: Signal<Signal<PaymentIntentEnvelope, ErrorEnvelope>.Event, Never> =
-      Signal.combineLatest(self.configureWithDataProperty.signal.skipNil(), checkoutId)
-        .takeWhen(self.applePayButtonTappedSignal)
-        .switchMap { initialData, checkoutId in
-          let projectId = initialData.project.graphID
-          let pledgeTotal = initialData.total
-
-          return stripeIntentService.createPaymentIntent(
-            for: projectId,
-            checkoutId: checkoutId,
-            pledgeTotal: pledgeTotal
-          )
-          .materialize()
-        }
-
-    let newPaymentIntentForApplePayError: Signal<ErrorEnvelope, Never> = createPaymentIntentForApplePay
-      .errors()
-
-    let newPaymentIntentForApplePay: Signal<String, Never> = createPaymentIntentForApplePay
-      .values()
-      .map { $0.clientSecret }
-
-    self.goToApplePayPaymentAuthorization = self
-      .configureWithDataProperty
-      .signal
-      .skipNil()
-      .combineLatest(with: newPaymentIntentForApplePay)
-      .map { (
-        data: PostCampaignCheckoutData,
-        paymentIntent: String
-      ) -> PostCampaignPaymentAuthorizationData? in
-        let baseReward = data.baseReward
-
-        return PostCampaignPaymentAuthorizationData(
-          project: data.project,
-          hasNoReward: baseReward.isNoReward,
-          subtotal: baseReward.isNoReward ? baseReward.minimum : calculateAllRewardsTotal(
-            addOnRewards: data.rewards,
-            selectedQuantities: data.selectedQuantities
-          ),
-          bonus: data.bonusAmount ?? 0,
-          shipping: data.shipping?.total ?? 0,
-          total: data.total,
-          merchantIdentifier: Secrets.ApplePay.merchantIdentifier,
-          paymentIntent: paymentIntent
-        )
-      }
-      .skipNil()
+    /// See ApplePayCheckoutViewModel.swift
 
-    let validateCheckoutWithApplePay = Signal.combineLatest(
-      newPaymentIntentForApplePay,
-      checkoutId,
-      self.applePayPaymentMethodIdSignal.signal
+    let applePayErrors = Signal.merge(
+      self.applePayViewModel.createNewPaymentIntentError,
+      self.applePayViewModel.validateCheckoutError
```

Comment:
Nit: I kind of think having these two separate error outputs makes the PostCampaignCheckoutViewModel a little too responsible for knowing how the applePayViewModel works. What do you think of combining these? Or, alternatively, having one ex `silentError` and one `showError`? (Assuming I'm reading the code right and that not telling the user about the payment intent error is intentional?)